src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -78,7 +79,7 @@ public PatternHelper( final ItemStack is, final World w )
 
 		this.canSubstitute = this.isCrafting && encodedValue.getBoolean( "substitute" );
 		this.patternItem = is;
-		this.pattern = AEItemStack.create( is );
+		this.pattern = AEItemStack.fromItemStack( is );
 
 		final List<IAEItemStack> in = new ArrayList<>();
 		final List<IAEItemStack> out = new ArrayList<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "out" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)